### PR TITLE
Fix cmake installation paths

### LINKED ISSUE #394: How to enable Translation
I through [AUR](https://aur.archlinux.org/packages/gittyup) and Flatpak installation Gittyup, after opening can only see English, can't see the Chinese translation, I need to do?（我通过 [AUR](https://aur.archlinux.org/packages/gittyup) 和 Flatpak 安装 Gittyup，打开后都只能看到英文，看不到中文翻译，我需要做些什么？）

### LINKED ISSUE #467: TreeView test no longer pass on Arch Linux
Hello, with v1.2.0 all test passed; but since v1.2.1 TreeView fail, haven't hunted the responsible commit yet

```
QDEBUG : TestTreeView::fileMergeCrash() Name:  "Unstaged"  Row:  0 :  "security_configs" ; Count:  0
QDEBUG : TestTreeView::fileMergeCrash() Name:  "Unstaged" , countCollapsed(): 0
QDEBUG : TestTreeView::fileMergeCrash() Name:  "Unstaged" Current Collapsed:  0 ; New Collapsed:  -1
TreeView: /usr/src/debug/gittyup-git/Gittyup/src/ui/TreeView.cpp:169: void TreeView::setCollapseCount(int): Assertion `value >= 0' failed.
```

Building the package using the AUR pkgbuild gittyup-git, attached there is the check log of makepkg

[gittyup-git-1.2.1.r0.gf364ff0-1-x86_64-check.log](https://github.com/Murmele/Gittyup/files/10777894/gittyup-git-1.2.1.r0.gf364ff0-1-x86_64-check.log)

### PR DESCRIPTION
**Pull request by Murmele**
fixes #394
fixes Bug introduced with #439
fixes #467 

- fixing installation path of the icons
Tested on:
- [x] Linux Flatpak (only locale are not working there yet)
- [x] Arch 
- [x] Windows
- [x] MacOS

### CODE REVIEW

**Comment by Murmele** [comment]
settings icons like this might not work anymore, because they are anymore in the correct folder used in the application:
```
#elif defined(Q_OS_LINUX)
  QIcon icon;
  icon.addPixmap(QPixmap(":/Gittyup.iconset/icon_16x16.png"));
  icon.addPixmap(QPixmap(":/Gittyup.iconset/icon_32x32.png"));
  icon.addPixmap(QPixmap(":/Gittyup.iconset/icon_64x64.png"));
  icon.addPixmap(QPixmap(":/Gittyup.iconset/icon_128x128.png"));
  setWindowIcon(icon);
#endif
```

**Review by alerque — APPROVED** [review]
Sorry I see I was tagged on this some time ago and never saw it to review. I gave the diff a quick read and everything seems to make sense, but at the moment (and for probably a few weeks) I don't have the extra bandwidth to setup some build tests and actually test drive the changes. Sorry but life just piles things on sometimes :shrug:

**Comment by Murmele** [comment]
@alerque currently I am working to get a clean build and installing everything to the correct folder. For this I have then the problem that libgit2 would be also installed into /usr/lib, which is not possible if there is another package installed.
So I think we need to create our own patched libgit2 version and installing it to another folder

https://stackoverflow.com/questions/75232118/cmake-set-target-install-directory

**Comment by alerque** [comment]
Yes if this is still reliant on a patched libgit2 that must be installed to a different name or location. Long term I would suggest distros are going to be a lot happier if this can be built against the system libgit2 libraries. The same goes for most 3rd party libraries that provide shared modules.

**Comment by Murmele** [comment]
> Yes if this is still reliant on a patched libgit2 that must be installed to a different name or location. Long term I would suggest distros are going to be a lot happier if this can be built against the system libgit2 libraries. The same goes for most 3rd party libraries that provide shared modules.

Yes I agree with you. Currently I have a few pr on libgit2 but nobody looked at them

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -308,6 +308,13 @@ jobs:
         with:
           path: build/release/VERSION.txt
           name: Gittyup-VERSION
+          
+      - name: Test2
+        if: matrix.env.ninja_platform != 'win' && matrix.env.ninja_platform != 'mac'
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          working-directory: build/release/tests
```
Comment (Murmele) [inline .github/workflows/build.yml:316]:
```suggestion
          working-directory: build/release/test
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -308,6 +308,13 @@ jobs:
         with:
           path: build/release/VERSION.txt
           name: Gittyup-VERSION
+          
+      - name: Test2
+        if: matrix.env.ninja_platform != 'win' && matrix.env.ninja_platform != 'mac'
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          working-directory: build/release/test
```
Comment (Murmele) [inline .github/workflows/build.yml:316]:
```suggestion
          working-directory: build/release
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -308,6 +308,13 @@ jobs:
...
+          run: ./init_repo
```
Comment (Murmele) [inline .github/workflows/build.yml:317]:
```suggestion
          run: ./test/init_repo
```

**Comment by Murmele** [comment]
Working ARCH PKGBUILD
```
# Maintainer: Fabio 'Lolix' Loli <lolix@disroot.org>
# Contributor: Tim Schumacher <timschumi@gmx.de>
# Contributor: KillWolfVlad <github.com/KillWolfVlad>
# Contributor: WaveHack <email@wavehack.net>
# Contributor: Whovian9369 <Whovian9369@gmail.com>
# Contributor: Martin Marmsoler <martin.marmsoler@gmail.com>

pkgname=gittyup-git
pkgver=1.2.2.r82.gc338a36
pkgrel=1
pkgdesc="Graphical Git client (GitAhead fork)"
url="https://github.com/Murmele/Gittyup"
arch=(x86_64)
license=(MIT)
depends=(qt5-base hunspell lua cmark pcre libssh2)
makedepends=(git cmake ninja qt5-tools qt5-translations ) #libgit2 libgnome-keyring
optdepends=('git-lfs: git-lfs support'
            'libgnome-keyring: for GNOME Keyring for auth credentials'
            'qt5-translations: translations')
provides=(gittyup)
conflicts=(gittyup)
source=("git+https://github.com/Murmele/Gittyup.git"
        "Gittyup-libgit2::git+https://github.com/stinb/libgit2.git"
        "git+https://github.com/kuba--/zip.git"
        "git+https://github.com/ScintillaOrg/lexilla.git"
        "git+https://github.com/orbitalquark/scintillua.git")
sha256sums=('SKIP'
            'SKIP'
            'SKIP'
            'SKIP'
            'SKIP')

pkgver() {
  cd "${srcdir}/Gittyup"
  git describe --long --tags --exclude latest | sed 's/^gittyup_v//;s/\([^-]*-g\)/r\1/;s/-/./g'
}

prepare() {
  cd "${srcdir}/Gittyup"

  git submodule init
  git config submodule.dep/libgit2/libgit2.url "${srcdir}/Gittyup-libgit2"
  git config submodule.dep/git/git.update none
  git config submodule.dep/cmark/cmark.update none
  git config submodule.dep/libssh2/libssh2.update none
  git config submodule.dep/openssl/openssl.update none
  git config submodule.dep/hunspell/hunspell.update none
  git config submodule.test/dep/zip.url "${srcdir}/zip"
  git config submodule.dep/scintilla/lexilla.url "${srcdir}/lexilla"
  git config submodule.dep/scintilla/scintillua.url "${srcdir}/scintillua"
  git -c protocol.file.allow=always submodule update

  [[ -d build ]] || mkdir build
}

build() {
  cd "${srcdir}/Gittyup/build"
  cmake -G Ninja .. -Wno-dev \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr/lib/gittyup \
    -DENABLE_REPRODUCIBLE_BUILDS=ON \
    -DGENERATE_APPDATA=ON \
    -DBUILD_SHARED_LIBS=OFF \
    -DUSE_SYSTEM_CMARK=ON \
    -DUSE_SYSTEM_GIT=ON \
    -DUSE_SYSTEM_HUNSPELL=ON \
    -DUSE_SYSTEM_LIBSSH2=ON \
    -DUSE_SYSTEM_LUA=ON \
    -DUSE_SYSTEM_OPENSSL=ON \
    -DUSE_SYSTEM_QT=ON

  ninja
}

check() {
  cd "${srcdir}/Gittyup"
  ninja -C build check
}

package() {
  cd "${srcdir}/Gittyup"
  DESTDIR="${pkgdir}" ninja -C build install

  install -d "${pkgdir}/usr/bin"
  ln -s /usr/lib/gittyup/bin/Gittyup "${pkgdir}/usr/bin/gittyup"
  ln -s /usr/lib/gittyup/bin/Gittyup "${pkgdir}/usr/bin/Gittyup"
  install -d "${pkgdir}/usr/share/applications"
  ln -s /usr/lib/gittyup/share/applications/Gittyup.desktop "${pkgdir}/usr/share/applications/Gittyup.desktop"
  
  for i in 16 32 64 128 256 512
  do
    install -d "${pkgdir}/usr/share/icons/hicolor/${i}x${i}/apps"
    ln -s /usr/lib/gittyup/share/icons/hicolor/${i}x${i}/apps/Gittyup.png "${pkgdir}/usr/share/icons/hicolor/${i}x${i}/apps/Gittyup.png"
  done
  install -d "${pkgdir}/usr/share/icons/hicolor/scalable/apps"
  ln -s /usr/lib/gittyup/share/icons/hicolor/scalable/apps/Gittyup.svg "${pkgdir}/usr/share/icons/hicolor/scalable/apps/Gittyup.svg"

  rm -rf "${pkgdir}/usr/lib/gittyup/"*.so.*
  rm -rf "${pkgdir}/usr/lib/gittyup/include"
  rm -rf "${pkgdir}/usr/lib/gittyup/lib"
}
```

**Comment by Murmele** [comment]
grafik

File: rsrc/linux/com.github.Murmele.Gittyup.desktop [hunk 4]
```diff
@@ -4,7 +4,7 @@ Name=Gittyup
 Comment=Graphical Git client for Windows, Linux and macOS
 Exec=Gittyup
```
Comment (alerque) [inline rsrc/linux/com.github.Murmele.Gittyup.desktop:5]: At least on *nix systems the installed executable that ends up in `$PATH` should be lower case `gittyup`. The Label in the desktop file can be upper case for pretty GUI users, but launching it from a terminal or run prompt should use lower case.

**Review by alerque — APPROVED** [review]
Without having tested it, reviewing the code here is looking good conceptually.

File: rsrc/linux/com.github.Murmele.Gittyup.desktop (hunk 4 above)
Comment (Murmele) [reply rsrc/linux/com.github.Murmele.Gittyup.desktop:5]: Thanks for the comment, I will check out that

**Comment by Murmele** [comment]
Updated pkgbuild with lowercase executable
````
# Maintainer: Fabio 'Lolix' Loli <lolix@disroot.org>
# Contributor: Tim Schumacher <timschumi@gmx.de>
# Contributor: KillWolfVlad <github.com/KillWolfVlad>
# Contributor: WaveHack <email@wavehack.net>
# Contributor: Whovian9369 <Whovian9369@gmail.com>
# Contributor: Martin Marmsoler <martin.marmsoler@gmail.com>

pkgname=gittyup-git
pkgver=1.2.2.r82.gc338a36
pkgrel=1
pkgdesc="Graphical Git client (GitAhead fork)"
url="https://github.com/Murmele/Gittyup"
arch=(x86_64)
license=(MIT)
depends=(qt5-base hunspell lua cmark pcre libssh2)
makedepends=(git cmake ninja qt5-tools qt5-translations ) #libgit2 libgnome-keyring
optdepends=('git-lfs: git-lfs support'
            'libgnome-keyring: for GNOME Keyring for auth credentials'
            'qt5-translations: translations')
provides=(gittyup)
conflicts=(gittyup)
source=("git+https://github.com/Murmele/Gittyup.git"
        "Gittyup-libgit2::git+https://github.com/stinb/libgit2.git"
        "git+https://github.com/kuba--/zip.git"
        "git+https://github.com/ScintillaOrg/lexilla.git"
        "git+https://github.com/orbitalquark/scintillua.git")
sha256sums=('SKIP'
            'SKIP'
            'SKIP'
            'SKIP'
            'SKIP')

pkgver() {
  cd "${srcdir}/Gittyup"
  git describe --long --tags --exclude latest | sed 's/^gittyup_v//;s/\([^-]*-g\)/r\1/;s/-/./g'
}

prepare() {
  cd "${srcdir}/Gittyup"

  git submodule init
  git config submodule.dep/libgit2/libgit2.url "${srcdir}/Gittyup-libgit2"
  git config submodule.dep/git/git.update none
  git config submodule.dep/cmark/cmark.update none
  git config submodule.dep/libssh2/libssh2.update none
  git config submodule.dep/openssl/openssl.update none
  git config submodule.dep/hunspell/hunspell.update none
  git config submodule.test/dep/zip.url "${srcdir}/zip"
  git config submodule.dep/scintilla/lexilla.url "${srcdir}/lexilla"
  git config submodule.dep/scintilla/scintillua.url "${srcdir}/scintillua"
  git -c protocol.file.allow=always submodule update

  [[ -d build ]] || mkdir build
}

build() {
  cd "${srcdir}/Gittyup/build"
  cmake -G Ninja .. -Wno-dev \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr/lib/gittyup \
    -DENABLE_REPRODUCIBLE_BUILDS=ON \
    -DGENERATE_APPDATA=ON \
    -DBUILD_SHARED_LIBS=OFF \
    -DUSE_SYSTEM_CMARK=ON \
    -DUSE_SYSTEM_GIT=ON \
    -DUSE_SYSTEM_HUNSPELL=ON \
    -DUSE_SYSTEM_LIBSSH2=ON \
    -DUSE_SYSTEM_LUA=ON \
    -DUSE_SYSTEM_OPENSSL=ON \
    -DUSE_SYSTEM_QT=ON

  ninja
}

check() {
  cd "${srcdir}/Gittyup"
  ninja -C build check
}

package() {
  cd "${srcdir}/Gittyup"
  DESTDIR="${pkgdir}" ninja -C build install

  install -d "${pkgdir}/usr/bin"
  ln -s /usr/lib/gittyup/bin/gittyup "${pkgdir}/usr/bin/gittyup"
  install -d "${pkgdir}/usr/share/applications"
  ln -s /usr/lib/gittyup/share/applications/gittyup.desktop "${pkgdir}/usr/share/applications/gittyup.desktop"
  
  for i in 16 32 64 128 256 512
  do
    install -d "${pkgdir}/usr/share/icons/hicolor/${i}x${i}/apps"
    ln -s /usr/lib/gittyup/share/icons/hicolor/${i}x${i}/apps/gittyup.png "${pkgdir}/usr/share/icons/hicolor/${i}x${i}/apps/gittyup.png"
  done
  install -d "${pkgdir}/usr/share/icons/hicolor/scalable/apps"
  ln -s /usr/lib/gittyup/share/icons/hicolor/scalable/apps/gittyup.svg "${pkgdir}/usr/share/icons/hicolor/scalable/apps/gittyup.svg"

  rm -rf "${pkgdir}/usr/lib/gittyup/"*.so.*
  rm -rf "${pkgdir}/usr/lib/gittyup/include"
  rm -rf "${pkgdir}/usr/lib/gittyup/lib"
}
````

File: rsrc/linux/com.github.Murmele.Gittyup.desktop (hunk 4 above)
Comment (Murmele) [reply rsrc/linux/com.github.Murmele.Gittyup.desktop]: This is now done in the latest build.

**Comment by Murmele** [comment]
@alerque @FabioLolix I got the Arch build ready, but for some reason I am not able to get a dark theme. Do you have the same problem?

**Pull request merged**